### PR TITLE
upgrade framework, upgrade shaders to new spir-v

### PR DESCRIPTION
**Pull request by WebFreak001**
note: I tested these things on top of #320 

I haven't tried if all shaders work though, I don't know how to run the visual tests.

### CODE REVIEW

**Comment by andy840119** [comment]
image

@WebFreak001 Thanks for your contribution! You saved my time 🎉 
Some of the test case are broken, but it still able to opened in the visual tests 🤔 
I have the same issue when trying to upgrade the shader in this morning 😢

**Comment by WebFreak001** [comment]
I noticed some shader code in the tests which I didn't upgrade yet. Gonna sleep for now, maybe gonna look at it tomorrow

**Comment by andy840119** [comment]
And notice that i might open another PR because uniform in the shader might designed like this:

```glsl
// see the demo: https://www.geeks3d.com/20101029/shader-library-pixelation-post-processing-effect-glsl/
#include "sh_Utils.h"

layout(location = 2) in mediump vec2 v_TexCoord;

// will pass the value in the CustomizedShaderBufferedDrawNode
layout(std140, set = 0, binding = 0) uniform m_SharedParameters
{
    mediump vec2 g_TexSize;
    float g_InflationPercentage;
};

// will pass the value in the customized shader. might have the same name for every shader? not very sure
layout(std140, set = 2, binding = 0) uniform m_PixelParameters
{
    mediump vec2 g_Size;
};

layout(set = 1, binding = 0) uniform lowp texture2D m_Texture;
layout(set = 1, binding = 1) uniform lowp sampler m_Sampler;

layout(location = 0) out vec4 o_Colour;

void main(void) 
{ 
    vec2 separaorParts = g_TexSize / (g_Size * g_InflationPercentage);
    vec2 uv = v_TexCoord;
    uv = uv * separaorParts;
    uv = floor(uv);
    uv = uv / separaorParts;
    o_Colour = toSRGB(texture(sampler2D(m_Texture, m_Sampler), uv));
}
```

**Comment by andy840119** [comment]
@WebFreak001 close this issue because create another PR for the minimum required change only.
Feel free to push to the #322 or create another PR based on that branch.

File: osu.Framework.Font/Graphics/Shaders/RepeatMovingBackgroundShader.cs [hunk 1]
```diff
@@ -2,52 +2,64 @@
...
+
     public float Mix { get; set; } = 1f;
 
     public override void ApplyValue()
     {
         if (Texture == null)
             return;
 
+        // where this 1 is from? I have no clue.
         Texture.Bind(1);
```
Comment (andy840119) [inline osu.Framework.Font/Graphics/Shaders/RepeatMovingBackgroundShader.cs:49]: Follow the change in the https://github.com/ppy/osu-framework/commit/9ec6711a909cf36ad7c82f12405a20071bf975a5#diff-ec8681ce3076c551c70eeebdc49697c9efb11b7d6f1e10bf0240235dbff7f7d3R122-R138

**Pull request closed without merging**